code/modules/mob/living/carbon/human/species.dm

```diff
@@ -428,19 +428,19 @@ GLOBAL_LIST_EMPTY(features_by_species)
 	// Drop the items the new species can't wear
 	if((AGENDER in species_traits))
 		C.gender = PLURAL
-	for(var/slot_id in no_equip)
-		var/obj/item/thing = C.get_item_by_slot(slot_id)
-		if(thing && (!thing.species_exception || !is_type_in_list(src,thing.species_exception)))
-			C.dropItemToGround(thing)
 	if(C.hud_used)
 		C.hud_used.update_locked_slots()
 
-
 	C.mob_biotypes = inherent_biotypes
 
 	replace_body(C, src)
 	regenerate_organs(C, old_species, visual_only = C.visual_only_organs)
 
+	for(var/obj/item/equipped_item in C.get_equipped_items(include_pockets = TRUE))
+		var/equipped_item_slot = C.get_slot_by_item(equipped_item)
+		if(!can_equip(equipped_item, equipped_item_slot, H = C,  bypass_equip_delay_self = TRUE))
+			C.dropItemToGround(equipped_item)
```

Comment:
There's a few problems here:

1- ``get_equipped_items`` gets every single item on a mob, but ``get_slot_by_item`` doesn't take pockets/suit storage into account, so they end up being null, we don't actually have to check pockets because we already check the jumpsuit/exosuit, and if those come off, their items go off with them.
2- can_equip returns FALSE if you already have a piece of clothing in that slot, and since we already have a full set of clothes, it causes us to drop everything.

Lastly, clothes have ``slot_flags`` which you can take advantage of here, bypassing the need for ``get_slot_by_item``.

Reply:
adjusted this